gpkit/model.py

```diff
@@ -367,16 +314,15 @@ def solve_pass(i):
                 this_pass.update(linked)
                 constants_ = constants
                 constants_.update(this_pass)
-                program, mmaps = self.formProgram(programType, posynomials,
-                                                  constants_, verbosity)
-
+                program, unsubbed.mmaps = form_program(
+                    programType, signomials, constants_, verbosity)
                 try:
                     if programType == "gp":
                         result = program.solve(*args, **kwargs)
                     elif programType == "sp":
                         result = program.localsolve(*args, **kwargs)
```

Comment:
the conditional inside this `try` makes me further think that `form_program` should just return the parts needed to call the constructor (which would be done here).